drivers/dma/dma_mcux_edma.c

```diff
@@ -120,14 +120,14 @@ static bool data_size_valid(const size_t data_size)
 static void nxp_edma_callback(edma_handle_t *handle, void *param, bool transferDone,
 			      uint32_t tcds)
 {
-	int ret = 1;
+	int ret = -EIO;
 	struct call_back *data = (struct call_back *)param;
 	uint32_t channel = handle->channel;
 
 	if (transferDone) {
 		/* DMA is no longer busy when there are no remaining TCDs to transfer */
 		data->busy = (handle->tcdPool != NULL) && (handle->tcdUsed > 0);
-		ret = 0;
+		ret = DMA_STATUS_COMPLETE;
```

Comment:
Shouldn't the `data->busy` be checked before setting `ret` to `DMA_STATUS_COMPLETE` or `DMA_STATUS_BLOCK`?

Reply:
This is effectively what was being done before with only the status codes being changed, if the driver has a bug in this regard please file an issue and assign to one of the NXP maintainers (e.g. dleach)